jobmanager/JobQueue.h

```diff
@@ -59,7 +60,30 @@ class JobQueue
     {
         concurrency::Semaphore sema;
         addJob<StopJob>(sema, _running);
-        sema.wait();
+
+        // Wait 1ms to this semaphore to be released, if it's not
+        // released after that we will check if this a worker thread
+        // and yield it to process other jobs that jobManager may have.
+        // If it is not an worker thread then will wait forever until semaphore is released
+        if (!sema.wait(1))
```

Comment:
why do the wait of 1ms at all if we are workerthread? If this is a workerthread we can just as well go on with the processing of jobs. No one should try add jobs to this queue after ~JobQueue has started to execute.

Reply:
Yeah, I thought the same when I was implementing but I decided to give a chance to execute the StopJob with a small wait. But I changes to not wait if there are jobs to run like you suggested